cmd/dep/status.go

```diff
@@ -643,3 +648,140 @@ func collectConstraints(ptree pkgtree.PackageTree, p *dep.Project, sm gps.Source
 	// TODO
 	return map[string][]gps.Constraint{}
 }
+
+type projectStatus struct {
+	Project               string
+	Version               string
+	Constraints           []string
+	Source                string
+	AltSource             string
+	PubVersions           map[string][]string
+	Revision              string
+	LatestAllowed         string
+	SourceType            string
+	Packages              []string
```

Comment:
The comment about `ProjectImports` and `PackageImports` made me realize that it would also be worth listing the ignores that are currently affecting this project - both what packages within it are ignored (that would otherwise actually be imported), and which of its imports are ignored.

This has the potential to be a little tricky, as if there's a hash mismatch (or if the user manually messed with `Gopkg.lock`), then it's possible that there may be some packages listed in `Gopkg.lock` that actually should be ignored. There's also a fair bit of work involved in actually pulling out the ignoreds, so i think it'll be fine for us to defer this to a later PR.

i've updated the spec with new lines and some comments explaining this, so we can discuss/dissect there; just noting it here.